Fix unsigned orders modal

bthaile opened the pull request:
fixes https://github.com/AugurProject/augur/issues/3286
fixes https://github.com/AugurProject/augur/issues/3308
associated with https://github.com/AugurProject/augur/issues/3285

change generate transaction id algo to use sha256 so the transaction parameters can be larger. 
updated lots of types.
moved calculate liquidity to reducer, there was a race condition doing the calculations outside of the reducer.

bthaile committed:
checkpoint

bthaile committed:
getting types straightened out

bthaile committed:
continuing to get types worked out.

bthaile committed:
Merge branch 'master' into fix-unsigned-orders

bthaile committed:
fixed merge conflict

phoebemirman committed:
fix bug where pending in markets list were getting mixed up

bthaile committed:
get single transactions working from unsigned transactions modal

bthaile committed:
fixed hashing string for tx id

phoebemirman commented on packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:
```diff
@@ -267,15 +274,15 @@ export function createMarketRetry(market: CreateMarketData) {
     minPrice: market.txParams._prices && market.txParams._prices[0],
     maxPrice: market.txParams._prices && market.txParams._prices[1],
-    tickSize: market.txParams._numTicks,
+    numTicks: market.txParams._numTicks,
```
don't you need to change the other createMarket call in submit-new-market to send numTicks instead of tickSize?

bthaile replied:
it works

phoebemirman reviewed (commented):
when i add liquidity, then go back and switch market types, the total DAI cost at the end doesn't update.

phoebemirman committed:
Merge branch 'master' into fix-unsigned-orders

phoebemirman committed:
Merge branch 'master' into fix-unsigned-orders

phoebemirman committed:
Merge branch 'market-row-fix' into fix-unsigned-orders

phoebemirman committed:
only show liquidity msg for succesful markets

phoebemirman committed:
Merge branch 'market-row-fix' into fix-unsigned-orders

phoebemirman committed:
Merge branch 'fix-unsigned-orders' of https://github.com/AugurProject/augur into fix-unsigned-orders

phoebemirman reviewed (commented):
this isn't working:
1. create a market with liquidity
2. it fails
3. retry
4. liquidity shows up

not sure if it was broken on purpose

bthaile committed:
fix numTick tickSize issue, for retry

bthaile committed:
Merge branch 'fix-unsigned-orders' of https://github.com/AugurProject/augur into fix-unsigned-orders

phoebemirman reviewed (commented):
liquidity isn't clearing from local storage after i submit an order

bthaile committed:
fix liquidity cost calcuations when user changes market type if they have existing orders

phoebemirman reviewed (commented):
- [x] liquidity shows up for all market types after first market creation try
- [x] liquidity shows up for all market types after second market creation try
- [x] total liquidity amount is correct after going back and changing market type
- [ ] liquidity modal has right amounts and information
- [ ] after submitting liquidity successfully, the liquidity goes away
- [x] after market fails, then you refresh, market shows, then after a successful retry the correct liquidity shows

bthaile committed:
fixed one thing with tx pending id creation

bthaile committed:
fixed positioning of parameters to get same hash value for new market as TX params

bthaile commented:
found a bug with creating scalar markets `tickSize` isn't on newMarket object. 

fixed bug with scalar markets retry. Fixed the tickSize/numTicks issue.

 *  liquidity modal has right amounts and information
 * after submitting liquidity successfully, the liquidity goes away
These will be done on a different PR

bthaile committed:
Merge branch 'master' into fix-unsigned-orders

bthaile committed:
fixed issue with numTicks and tickSize properties when retrying create market

bthaile committed:
fixed scalar market retry, price was mul by QUINTILLION twice

bthaile committed:
Merge branch 'fix-unsigned-orders' of https://github.com/AugurProject/augur into fix-unsigned-orders

bthaile committed:
Merge branch 'master' into fix-unsigned-orders

bthaile committed:
move scalar liquidity to correct outcome

bthaile committed:
Merge branch 'fix-unsigned-orders' of https://github.com/AugurProject/augur into fix-unsigned-orders

phoebemirman committed:
Merge branch 'master' into fix-unsigned-orders
